django/test/testcases.py

```diff
@@ -756,6 +756,18 @@ def assertXMLNotEqual(self, xml1, xml2, msg=None):
                 standardMsg = '%s == %s' % (safe_repr(xml1, True), safe_repr(xml2, True))
                 self.fail(self._formatMessage(msg, standardMsg))
 
+    def assertFormValid(self, form, msg=None):
+
+        if not form.is_valid():
+            standardMsg = form.errors.as_json()
+            self.fail(self._formatMessage(msg, standardMsg))
+
+    def assertFormNotValid(self, form, msg=None):
+
+        if form.is_valid():
+            standardMsg = 'assertFormNotValid validated succesfully'
```

Comment:
If we wish to implement the complement of this, here's some nicer pseudocode:
```
def assertFormNotValid(self, form, expected_errors=None, msg=None):
    to_report = []
    if form.is_valid():
        to_report.append('form was valid')
    else:
        for error in expected_errors:
            if not form.has_error(*error):
                to_report.append('missing %s' % error)
    if to_report:
        self.fail(to_report)
```

The idea is to optionally allow you to pass in a list of `(field, code)` tuples (this should be more flexible to the fact code is optional actually), which will be checked as having thrown errors. Extra errors are ignored (design decision?).